kafka-ui-react-app/src/components/Schemas/Edit/Edit.tsx

```diff
@@ -0,0 +1,179 @@
+  clusterName: ClusterName;
+  schemasAreFetched: boolean;
+  fetchSchemasByClusterName: (clusterName: ClusterName) => void;
+  updateSchema: (
+    latestSchema: SchemaSubject,
+    newSchema: string,
+    newSchemaType: SchemaType,
+    newCompatibilityLevel: CompatibilityLevelCompatibilityEnum,
+    clusterName: string,
+    subject: string
+  ) => Promise<void>;
+}
+
+export default function Edit({
+  subject,
+  schema,
+  clusterName,
+  schemasAreFetched,
+  fetchSchemasByClusterName,
+  updateSchema,
+}: EditProps) {
+  React.useEffect(() => {
+    if (!schemasAreFetched) fetchSchemasByClusterName(clusterName);
+  }, [clusterName, fetchSchemasByClusterName]);
+
+  const [newSchema, setNewSchema] = React.useState('');
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const { register, handleSubmit } = useForm<NewSchemaSubjectRaw>();
+  const getFormattedSchema = React.useCallback(
+    () => JSON.stringify(JSON.parse(schema.schema), null, '\t'),
+    [schema]
+  );
+
+  const handleSchemaChange = React.useCallback(
+    (e: string) => {
+      setNewSchema(e);
+    },
+    [setNewSchema]
+  );
+  const history = useHistory();
+  const onSubmit = React.useCallback(
+    async ({
+      schemaType,
+      compatibilityLevel,
+    }: {
+      schemaType: SchemaType;
+      compatibilityLevel: CompatibilityLevelCompatibilityEnum;
+    }) => {
+      try {
+        setIsSubmitting(true);
+        await updateSchema(
+          schema,
+          newSchema,
+          schemaType,
+          compatibilityLevel,
+          clusterName,
+          subject
+        );
+        history.push(clusterSchemaPath(clusterName, subject));
+      } catch (e) {
+        setIsSubmitting(false);
+      }
+    },
+    [schema, newSchema, register, clusterName, subject, updateSchema, history]
+  );
+
+  return (
+    <div className="section">
+      <div className="level">
+        <div className="level-item level-left">
+          <Breadcrumb
+            links={[
+              {
+                href: clusterSchemasPath(clusterName),
+                label: 'Schema Registry',
+              },
+              {
+                href: clusterSchemaPath(clusterName, subject),
+                label: subject,
+              },
+            ]}
+          >
+            Edit
+          </Breadcrumb>
+        </div>
+      </div>
+
+      {schemasAreFetched && !isSubmitting ? (
+        <div className="box">
+          <div className="columns">
+            <div className="column is-one-half">
+              <h4 className="title is-5 mb-2">Latest Schema</h4>
+              <JSONEditor
+                readonly
+                value={getFormattedSchema()}
+                name="latestSchema"
+              />
+            </div>
+            <div className="column is-one-half">
+              <h4 className="title is-5 mb-2">New Schema</h4>
+              <JSONEditor
+                value={newSchema || getFormattedSchema()}
```

Comment:
Should this also be a part of the form which takes `getFormattedSchema()` as an initial value?

Reply:
This is still not addressed.